lib/shape.ml

```diff
@@ -60,6 +60,28 @@ let translate dx dy shape =
 
 let show shapes = List.iter render_shape shapes
 
+let bi_to_uni x y = 
```

Comment:
Are you able to explain what this helper function does, please?

Reply:
Sure! Multiplying by `sin degrees` or `cos degrees` maps our coordinates to a signed(or as I like to think of it, bi-polar) range (-size to size), but our coordinate space is unsigned/unipolar. So the result of our rotation needs to be scaled back to that range in order for the rotation to be properly centered around the point we originally fed it, which is what this function does.

I definitely could have inlined this, but I thought it may be useful for other things, I know in my own creative coding library it and its complement have both been very useful :)

Reply:
Got it, thanks for the explanation. Sounds good to me.